contrib/format-fixedwidth/src/main/java/org/apache/drill/exec/store/fixedwidth/FixedwidthBatchReader.java

```diff
@@ -0,0 +1,193 @@
+    }
+
+    reader = new BufferedReader(new InputStreamReader(fsStream, Charsets.UTF_8));
+
+    return true;
+
+  }
+
+  @Override
+  public boolean next() { // Use loader to read data from file to turn into Drill rows
+
+    String line;
+
+    try {
+      line = reader.readLine();
+      RowSetLoader writer = loader.writer();
+
+      while (!writer.isFull() && line != null) {
+
+        writer.start();
+        parseLine(line, writer);
+        writer.save();
+
+        line = reader.readLine();
+      }
+    } catch (Exception e) {
+      throw UserException
+              .dataReadError(e)
+              .message("Failed to read input file: {}", split.getPath().toString())
+              .addContext(errorContext)
+              .addContext(e.getMessage())
+              .build(logger);
+    }
+    return (line != null);
+  }
+
+  @Override
+  public void close() {
+    try {
+      fsStream.close();
+      loader.close();
+    } catch (Exception e) {
+      throw UserException
+              .dataReadError(e)
+              .message("Failed to close input file: {}", split.getPath().toString())
+              .addContext(errorContext)
+              .addContext(e.getMessage())
+              .build(logger);
+    }
+  }
+
+  private TupleMetadata buildSchema(){
+    SchemaBuilder builder = new SchemaBuilder();
+
+    for (FixedwidthFieldConfig field : config.getFields()){
+      builder.addNullable(field.getFieldName(),field.getDataType());
+    }
+
+      return builder.buildSchema();
+  }
+
+  private void parseLine(String line, RowSetLoader writer) {
+    int i = 0;
+    TypeProtos.MinorType dataType;
+    String dateTimeFormat;
+    String value;
+
+    for (FixedwidthFieldConfig field : config.getFields()) {
+      value = line.substring(field.getStartIndex() - 1, field.getStartIndex() + field.getFieldWidth() - 1);
+
+      dataType = field.getDataType();
+      dateTimeFormat = field.getDateTimeFormat();
+      DateTimeFormatter formatter = DateTimeFormatter.ofPattern(dateTimeFormat, Locale.ENGLISH);
+
+      switch (dataType) {
+        case INT:
+          writer.scalar(i).setInt(Integer.parseInt(value));
+          break;
+        case VARCHAR:
+          writer.scalar(i).setString(value);
+          break;
+        case DATE:
+          LocalDate date = LocalDate.parse(value, formatter);
+          writer.scalar(i).setDate(date);
+          break;
+        case TIME:
+          LocalTime time = LocalTime.parse(value, formatter);
+          writer.scalar(i).setTime(time);
+          break;
+        case TIMESTAMP:
+          LocalDateTime ldt = LocalDateTime.parse(value,formatter);
+          ZoneId z = ZoneId.of( "America/Toronto" );
+          ZonedDateTime zdt = ldt.atZone( z );
+          Instant timeStamp = zdt.toInstant();
+          writer.scalar(i).setTimestamp(timeStamp);
+          break;
+        default:
+          throw new RuntimeException("Unknown data type specified in fixed width. Found data type " + dataType);
+
```

Comment:
Nit: Please remove excess whitespace here and elsewhere.

Reply:
Done